examples/src/main/java/com/mapbox/navigation/examples/core/MapboxBuildingHighlightActivity.kt

```diff
@@ -0,0 +1,374 @@
+             * user has selected.
+             */
+            binding.mapView.getGesturesPlugin().addOnMapClickListener { point ->
+                buildingHighlightApi.highlightBuilding(point)
+                false
+            }
+        }
+    }
+
+    private fun getMapboxAccessTokenFromResources(): String {
+        return getString(this.resources.getIdentifier("mapbox_access_token", "string", packageName))
+    }
+
+    private fun getMapCamera(): CameraAnimationsPlugin {
+        return binding.mapView.getCameraAnimationsPlugin()
+    }
+
+    private fun startSimulation(route: DirectionsRoute) {
+        mapboxReplayer.stop()
+        mapboxReplayer.clearEvents()
+        mapboxReplayer.pushRealLocation(this, 0.0)
+        val replayEvents = ReplayRouteMapper().mapDirectionsRouteGeometry(route)
+        mapboxReplayer.pushEvents(replayEvents)
+        mapboxReplayer.seekTo(replayEvents.first())
+        mapboxReplayer.play()
+    }
+
+    private fun updateCamera(location: Location) {
+        val mapAnimationOptionsBuilder = MapAnimationOptions.Builder()
+        mapAnimationOptionsBuilder.duration(1500L)
+        mapCamera.easeTo(
+            CameraOptions.Builder()
+                .center(Point.fromLngLat(location.longitude, location.latitude))
+                .bearing(location.bearing.toDouble())
+                .pitch(45.0)
+                .zoom(17.0)
+                .padding(EdgeInsets(1000.0, 0.0, 0.0, 0.0))
+                .build(),
+            mapAnimationOptionsBuilder.build()
+        )
+    }
+
+    private fun findRoute(origin: Point) {
+        val routeOptions = RouteOptions.builder()
+            .baseUrl(Constants.BASE_API_URL)
+            .user(Constants.MAPBOX_USER)
+            .profile(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
+            .geometries(DirectionsCriteria.GEOMETRY_POLYLINE6)
+            .requestUuid("")
+            .accessToken(getMapboxAccessTokenFromResources())
+            .coordinates(waypointsController.coordinates(origin))
+            .alternatives(true)
+            .build()
+        mapboxNavigation.requestRoutes(
+            routeOptions,
+            object : RoutesRequestCallback {
+                override fun onRoutesReady(routes: List<DirectionsRoute>) {
+                    mapboxNavigation.setRoutes(routes)
+                }
+
+                override fun onRoutesRequestFailure(
+                    throwable: Throwable,
+                    routeOptions: RouteOptions
+                ) {
+                    // no impl
+                }
+
+                override fun onRoutesRequestCanceled(routeOptions: RouteOptions) {
+                    // no impl
+                }
+            }
+        )
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = LayoutActivityTripprogressBinding.inflate(layoutInflater)
```

Comment:
Let's add a new layout file for this activity
`layout_activity_building_highlight` and then use `LayoutActivityBuildingHighlightBinding`